game/overlord/iso.cpp

```diff
@@ -793,6 +823,120 @@ u32 ProcessVAGData(IsoMessage* _cmd, IsoBufferHeader* buffer_header) {
 // TODO - UpdatePlayPos
 // TODO - CheckVAGStreamProgress
 
+u32 GetPlayPos();
+void UpdatePlayPos();
+void StopVAG(VagCommand* cmd);
+void PauseVAG(VagCommand* cmd);
+void UnpauseVAG(VagCommand* cmd);
+void CalculateVAGVolumes(u32 a1, u32 a2, u32 a3, int* a4, int* a5);
+void SetVAGVol();
+
+u32 GetPlayPos() {
```

Comment:
This function is one that we'll probably be interested in soon. When playing cutscenes, the animation player tries to stay in sync with the audio.  This function gets called once per frame on the IOP, and the result gets sent back to the EE.  Unfortunately it uses the `sceSd` library for some stuff, so it's not super simple.

That said, I'd guess that this returns the position in the stream in audio samples.

Reply:
Indeed, the libsd call here is sceSdGetAddr(SD_VA_NAX | gVoice) which will get the current location of the voice in SPU memory.